docs/shared/plugins/create-plugin.md

```diff
@@ -0,0 +1,148 @@
+
+### Generator Testing
+
+The generator spec file includes boilerplate to help get started with testing. This includes setting up an empty workspace.
+
+These tests should ensure that files within the tree (created with `createTreeWithEmptyWorkspace`) are in the correct place, and contain the right content.
+
+Full E2Es are supported (and recommended) and will run everything on the file system like a user would.
+
+## Executor
+
+The default executor is set up to just emit a console log. Some examples of what an executor can do are:
+
+- Support different languages, (Java, Go, Python, C#)
+- Compile new UI framework components
+- Deploy an app on a CDN
+- Publish to NPM
+- and many more!
+
+### Adding more executors
+
+To [add more executors](/plugins/executors/creating-custom-executors) to the plugin, run the following command:
+`nx generate @nrwl/nx-plugin:executor [executor] --project=[pluginName]`.
+
+This will scaffold out a new generator and update the necessary files to support it.
+
+### Executor testing
+
+The executor spec file contains boilerplate to run the default exported function from the executor.
+
+These tests should make sure that the executor is executing and calling the functions that it relies on.
+
+Full E2Es are supported (and recommended) and will run everything on the file system like a user would.
+
+## Testing your plugin
+
+One of the biggest benefits that the Nx Plugin package provides is support for E2E and unit testing.
+
+When the E2E app runs, a temporary E2E directory is created in the root of the workspace. This directory is a blank Nx workspace, and will have the plugin's built package installed locally.
+
+### E2E Testing file
+
+When the plugin is generated, a test file is created in the `my-plugin-e2e` app. Inside this test file, there are already tests for making sure that the executor ran, checking if directories are created with the `--directory` option, and checking if tags are added to the project configuration.
+
+We'll go over a few parts of a test file below:
+
+```typescript
+it('should create my-plugin', async (done) => {
+  const plugin = uniq('my-plugin');
+  ensureNxProject('@my-org/my-plugin', 'dist/packages/my-plugin');
+  await runNxCommandAsync(`generate @my-org/my-plugin:myPlugin ${plugin}`);
+
+  const result = await runNxCommandAsync(`build ${plugin}`);
+  expect(result.stdout).toContain('Executor ran');
+
+  done();
+});
+```
+
+- The `uniq` function creates a random name with the prefix and a random number.
+- The `ensureNxProject` is the function that will create the temporary directory. It takes two arguments, the plugin package name and the dist directory of when it's built.
+- The `runNxCommandAsync` will execute a `nx` command in the E2E directory.
+
+There are additional functions that the `@nrwl/nx-plugin/testing` package exports. Most of them are file utilities to manipulate and read files in the E2E directory.
```

Comment:
This file is pretty slim by default now, since there is not a default executor / generator to be able to test - see: https://github.com/nrwl/nx/pull/16168

I'd recommend shifting this to a "How to write e2e tests for plugins" section, or maybe adding an addendum that covers the helpers a bit more thoroughly